### PR TITLE
[DROOLS-7052] Refactor PMML junit 4 -> 5

### LINKED ISSUE #1454: Property place holders in Maven dependency `classifier` and `type` should be evaluated to their values
This can manifest as a dependency not having a version when the classifier/type are being used to find a managed dependency: 

Reported by [radcortez](https://github.com/radcortez):

Another one is No version provided for dependency io.netty:netty-transport-native-epoll, which seems to be caused by this:
https://github.com/apache/activemq-artemis/blob/2.19.0/artemis-server/pom.xml#L129-L138

### PR DESCRIPTION
**Pull request by gitgabrio**
@mariofusco @pibizza 
**JIRA**: 

[link](https://issues.redhat.com/browse/DROOLS-7052)

This PR
1)  migrates pmml tests from junit 4 to 5 (helper PR for https://github.com/kiegroup/drools/pull/4503)
2) clean up/remove unwanted inheritance of junit 4 inside kie-pmml
3) upgrade rewrite-maven-plugin version to fix https://github.com/openrewrite/rewrite/issues/1454
4) disable pmml-archetypes modules (not needed nor maintained anymore)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-scorecard/kie-pmml-models-drools-scorecard-evaluator/pom.xml [hunk 1]
```diff
@@ -42,11 +46,23 @@
       <groupId>org.kie</groupId>
       <artifactId>kie-pmml-compiler-api</artifactId>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>junit</groupId>
+          <artifactId>junit</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-scorecard/kie-pmml-models-drools-scorecard-evaluator/pom.xml:54]: This should not be necessary

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-scorecard/kie-pmml-models-drools-scorecard-evaluator/pom.xml [hunk 2]
```diff
@@ -42,11 +46,23 @@
...
     <dependency>
       <groupId>org.kie</groupId>
       <artifactId>kie-pmml-compiler-api</artifactId>
       <classifier>tests</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>junit</groupId>
+          <artifactId>junit</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-scorecard/kie-pmml-models-drools-scorecard-evaluator/pom.xml:65]: This should not be necessary

File: kie-pmml-trusty/pom.xml [hunk 3]
```diff
@@ -16,12 +16,18 @@
...
+  </properties>
+
   <modules>
     <module>kie-pmml-commons</module>
     <module>kie-pmml-compiler</module>
     <module>kie-pmml-evaluator</module>
-    <module>kie-pmml-models-archetype</module>
-    <module>kie-pmml-models-drools-archetype</module>
+<!--    <module>kie-pmml-models-archetype</module>-->
+<!--    <module>kie-pmml-models-drools-archetype</module>-->
```
Comment (danielezonca) [inline kie-pmml-trusty/pom.xml:30]: Can you please create a ticket to update/remove/fix them?

**Comment by gitgabrio** [comment]
@danielezonca @mariofusco Hi! Could you please merge this, squashing commits?  It helps a lot for reading following PRs Thanks!

**Review by danielezonca — APPROVED** [review]
LGTM just some minor comments

**Pull request merged**